cranelift-codegen/src/licm.rs

```diff
@@ -156,12 +155,25 @@ fn trivially_unsafe_for_licm(opcode: Opcode) -> bool {
         || opcode.writes_cpu_flags()
 }
 
+fn is_unsafe_load(inst_data: &InstructionData) -> bool {
+    match *inst_data {
+        InstructionData::Load { flags, .. } | InstructionData::LoadComplex { flags, .. } => {
+            !flags.readonly() || !flags.notrap()
+        }
+        _ => inst_data.opcode().can_load(),
+    }
+}
```

Comment:
This is now the same code as simple_gvn.rs' `is_load_and_not_readonly`. I'm not particular about the name here, but this would be a good point to factor out this logic into a shared helper function in cranelift/cranelift-codegen/src/ir/instructions.rs. It looks like we can factor out `trivially_safe_for_licm`/`trivially_safe_for_gvn` now too.